### PR TITLE
Added CPP Code for Solution-3 (*^▽^*)

### PR DESCRIPTION
**Pull request by Ishwarendra**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-712.mdx
Comment (envyaims) [inline solutions/usaco-712.mdx:200]: i recommend putting brackets around every statement

**Review by envyaims — APPROVED** [review]
looks great! dont forget to add yourself to the author list :)

File: solutions/usaco-712.mdx
Comment (Ishwarendra) [reply solutions/usaco-712.mdx:200]: > looks great! dont forget to add yourself to the author list :)

Hehe.. Thanks for telling

Comment (Ishwarendra) [reply solutions/usaco-712.mdx:200]: > i recommend putting brackets around every statement
Ok I will put them up

**Review by envyaims — APPROVED** [review]
no need to request another review after i approved lol

File: solutions/usaco-712.mdx
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:213]: over 80 chars

File: solutions/usaco-712.mdx
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:223]:
```suggestion
```

File: solutions/usaco-712.mdx
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:221]:
```suggestion
	std::cout << crossings.size() << endl;
```

File: solutions/usaco-712.mdx
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:169]: quick comment explaining what this means

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-712.mdx
Comment (Ishwarendra) [reply solutions/usaco-712.mdx:169]: Line 167 to 169? 
In question the length of string is fixed (26 * 2 = 56) so I decided to add a const variable representing total number of characters.

File: solutions/usaco-712.mdx
Comment (Ishwarendra) [reply solutions/usaco-712.mdx:213]: splitted into steps ✅

File: solutions/usaco-712.mdx
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:168]:
```suggestion
using namespace std;

```

File: solutions/usaco-712.mdx
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:177]:
```suggestion
	string recordings;
	cin >> recordings;
```

File: solutions/usaco-712.mdx
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:180]:
```suggestion
	vector<int> first_found(26, -1);
	set<pair<char, char>> crossings;
```

File: solutions/usaco-712.mdx
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:195]:
```suggestion
			set<char> appearances;
```

File: solutions/usaco-712.mdx
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:223]:
```suggestion
	cout << crossings.size() << "\n";
```

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]
ahh using namespace std

File: solutions/usaco-712.mdx
Comment (SansPapyrus683) [reply solutions/usaco-712.mdx:168]: hey op, all of these changes are optional, if you wanna qualify the namespace each time it's fine by me

Comment (Ishwarendra) [reply solutions/usaco-712.mdx:168]: f these changes are optional, if you wanna qualify the namespace each time it's fine by me



> hey op, all of these changes are optional, if you wanna qualify the namespace each time it's fine by me

i think it's ok to keep them.

**Review by Ishwarendra — COMMENTED** [review]
No issue with any changes done.

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**
